### PR TITLE
Do not run limitrange admission plugin on pod update requests

### LINKED ISSUE #62666: Bad interaction between finalizer, limit range and mutating web-hook prevents pod termination
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**: Pod stuck in terminating state.

**What you expected to happen**: Pod successfully cleaned up.

**How to reproduce it (as minimally and precisely as possible)**:
1. Set up a LimitRange. (Say "limits:
    - defaultRequest:
        cpu: 100m
      type: Container")
2. Install a mutating webhook on pod which creates an additional container on the pod (Eg. https://istio.io/docs/setup/kubernetes/sidecar-injection.html#automatic-sidecar-injection) without the limit range.
3. Create a Pod with foreground deletion finalizer.
4. Notice that you have a Pod with 2 containers. The 1 you created which now has a limit and the istio container which does not.
5. Try to delete the Pod.
6. The garbage collector tries to update the pod to remove the finalizer, however this triggers the limit range admission control to add the limit to the istio container. This then causes the API server to get the error "invalid: spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or `spec.tolerations`"
7. Notice the pod is now stuck in terminating.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.9.6

/sig api-machinery

### PR DESCRIPTION
**Pull request by jennybuckley**
/kind bug
/sig api-machinery

**What this PR does / why we need it**:
According to [this](https://github.com/kubernetes/kubernetes/blob/6ec1ebd6ce65b7b05f195551e89649428d8c2738/staging/src/k8s.io/api/core/v1/types.go#L2740-L2742) and [this](https://github.com/kubernetes/kubernetes/blob/6ec1ebd6ce65b7b05f195551e89649428d8c2738/staging/src/k8s.io/api/core/v1/types.go#L2733-L2734), Pod containers and initContainers cannot be added, removed, or updated. Because of this, it is unnecessary to run the limitrange plugin on pod update requests (The limitrange plugin on pods only mutates and validates containers and initContainers).

Trying to mutate containers or initContainers on a pod update request will always fail pod validation because those fields are immutable once the object is created. Validating containers or initContainers on a pod update request should never give a different result from what it returned when the pod object was created (for the same reason). Trying to validate pod updates could prevent updates to pods created when the limitrange plugin was not enabled or was enabled with different limits.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #62666

**Release note**:
```release-note
NONE
```

/cc @cheftako @caesarxuchao

### CODE REVIEW

**Comment by jennybuckley** [comment]
misspelled "unnecessary"

**Comment by jennybuckley** [comment]
@cheftako I think this must have been blocked on something before, but I don't remember what. I just rebased it and I think it could be ready to merge now. Can you PTAL?

**Review by cheftako — APPROVED** [review]
/lgtm

Do you know why there is so much common code between TestLimitRangerIgnoresSubresource and TestLimitRangerAdmitPod? The first test seems to be a strict subset of the second.

**Comment by jennybuckley** [comment]
@deads2k realized this never got the right approval to merge, can you please take a look at this change to the limitrange admission plugin?

**Pull request merged**
